src/git_platform/zcl_abapgit_git_url.clas.abap

```diff
@@ -81,4 +88,20 @@ CLASS ZCL_ABAPGIT_GIT_URL IMPLEMENTATION.
     ENDIF.
 
   ENDMETHOD.
+
+
+  METHOD validate_url.
+
+    DATA lv_provider TYPE string.
+
+    lv_provider = zcl_abapgit_url=>host( to_lower( iv_url ) ).
+
+    IF lv_provider CS 'gitlab.com'.
```

Comment:
Are you sure this is only gitlab behavior ? My feeling is that, in opposite, just the gitHUB allows omitting the `.git suffix`. not sure for 100% ... but `.git` suffix was there first ... so I suspect most providers just follow this tradition

Reply:
check bitbucket + azure devops, think that covers 90% of the market then

Reply:
Yeap ... ideally check each vendor if the allow omitting `.git`
And if not, the it's statistically `if NOT provider cs 'github.com'`

Reply:
bitbucket works without `.git`

Reply:
What about silently just adding a ".git"-suffix if it's not there? Then no provider-specific checks have to be added?

Reply:
I like the idea with auto suffix.

Reply:
that's no guarantee either... Azure DevOps with `.git`:

![image](https://user-images.githubusercontent.com/59966492/207072312-bba29bc4-674f-49f8-bc68-e386ac3b401b.png)